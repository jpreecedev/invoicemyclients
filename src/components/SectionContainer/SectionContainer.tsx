import React from 'react'
import clsx from 'clsx'

import styles from './styles.module.css'
import { HeaderSubheader, Header } from 'semantic-ui-react'

interface SectionContainerProps {
  id: string
  title: string
  subtitle?: string
  sectionClassNames?: string
}

const SectionContainer: React.FC<SectionContainerProps> = ({
  children,
  id,
  title,
  subtitle,
  sectionClassNames
}) => {
  return (
    <section id={id} className={clsx(styles.container, 'drop-shadow', sectionClassNames)}>
      <Header as="h2">
        {title}
        {subtitle && <HeaderSubheader>{subtitle}</HeaderSubheader>}
      </Header>

      {children}
    </section>
  )
}

export { SectionContainer }
