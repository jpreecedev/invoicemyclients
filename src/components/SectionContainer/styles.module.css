.container {
  width: 100%;
  max-width: 720px;
  margin: auto;
  padding: 32px;
}
