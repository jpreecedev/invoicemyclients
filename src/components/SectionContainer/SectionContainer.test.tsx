import React from 'react'
import { render } from '@testing-library/react'
import { SectionContainer } from './SectionContainer'

describe('<SectionContainer /> tests', () => {
  it('should match snapshot', () => {
    expect(
      render(
        <SectionContainer
          id="test"
          title="Test Container"
          subtitle="Test Sub Title"
          sectionClassNames="optional-classes"
        />
      ).container
    ).toMatchSnapshot()
  })
})
