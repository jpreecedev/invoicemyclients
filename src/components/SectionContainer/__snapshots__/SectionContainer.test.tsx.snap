// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainer /> tests should match snapshot 1`] = `
<div>
  <section
    class="container drop-shadow optional-classes"
    id="test"
  >
    <h2
      class="ui header"
    >
      Test Container
      <div
        class="sub header"
      >
        Test Sub Title
      </div>
    </h2>
  </section>
</div>
`;
